### PR TITLE
Create `Client` entity

### LINKED ISSUE #5: Create `Client` entity
### Description
Create `Client` entity

Infrastructure link - https://link.excalidraw.com/l/7x0yF4kXlsV/8ho7kx5CwvD

### Requirements
- project should be runnable

### Deliverables
- `Client` entity, 
- `Client` repository
- `Client` foundation service
- Fluent Valiation
- Add **temporary** seed data
- Add Migration
- Register all necessary infrastructure to DI

### PR DESCRIPTION
**Pull request by bunyodkamalov**
closes #5

### CODE REVIEW

**Review by botirjonabdumannonovv — APPROVED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/CommandHandlers/ClientGetByIdCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+using Feedback.Analyzer.Domain.Common.Commands;
+using Feedback.Analyzer.Domain.Common.Queries;
+using Feedback.Analyzer.Domain.Entities;
+
+namespace Feedback.Analyzer.Infrastructure.CommandHandlers;
+
+/// <summary>
+/// Command handler for retrieving a client by their unique identifier.
+/// </summary>
+public class ClientGetByIdCommandHandler(IClientService clientService) : ICommandHandler<ClientGetByIdCommand, Client?>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/CommandHandlers/ClientGetByIdCommandHandler.cs:12]: Change to query

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/CommandHandlers/ClientGetCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+using Feedback.Analyzer.Domain.Common.Commands;
+using Feedback.Analyzer.Domain.Common.Queries;
+using Feedback.Analyzer.Domain.Entities;
+
+namespace Feedback.Analyzer.Infrastructure.CommandHandlers;
+
+/// <summary>
+/// Command handler for retrieving a collection of clients based on specified criteria.
+/// </summary>
+public class ClientGetCommandHandler(IClientService clientService) : ICommandHandler<ClientGetCommand, IQueryable<Client>>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/CommandHandlers/ClientGetCommandHandler.cs:12]: Change to query handler

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Clients/Services/ClientService.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+using Feedback.Analyzer.Application.Clients.Services;
+using Feedback.Analyzer.Domain.Entities;
+using Feedback.Analyzer.Domain.Enums;
+using Feedback.Analyzer.Infrastucture.Common.Validators;
+using Feedback.Analyzer.Persistence.Repositories.Interfaces;
+using FluentValidation;
+
+namespace Feedback.Analyzer.Infrastucture.Clients.Services;
+
+public class ClientService(
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Clients/Services/ClientService.cs:12]: Add type docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Clients/Services/ClientService.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    ClientValidator clientValidator) : IClientService
+{
+    public IQueryable<Client> Get
+        (Expression<Func<Client, bool>>? predicate = default, bool asNoTracking = false)
+        => clientRepository.Get(predicate, asNoTracking);
+
+    public IQueryable<Client> Get
+        (ClientFilter clientFilter, bool asNoTracking = false)
+    {
+        return clientRepository.Get(asNoTracking: asNoTracking).Skip((int)((clientFilter.PageToken - 1) * clientFilter.PageSize))
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Clients/Services/ClientService.cs:23]: Use ApplyPagination extension

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Clients/Services/ClientService.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+/// <summary>
+/// Service implementation for managing client entities.
+/// </summary>
+public class ClientService(IClientRepository clientRepository) : IClientService
+{
+    public IQueryable<Client> Get(Expression<Func<Client, bool>>? predicate = default, QueryOptions queryOptions = default)
+        => clientRepository.Get(predicate, queryOptions);
+
+    public IQueryable<Client> Get(ClientFilter clientFilter, QueryOptions queryOptions = default)
+        => clientRepository.Get(queryOptions: queryOptions).Skip((int)((clientFilter.PageToken - 1) * clientFilter.PageSize)).Take((int)(clientFilter.PageSize));
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Clients/Services/ClientService.cs:20]: Use ApplyPagination extension

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Common/Validators/ClientValidator.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        RuleSet(EntityEvent.OnCreate.ToString(), () => 
+        {
+            RuleFor(client => client.FirstName).NotEmpty().MaximumLength(64);
+            
+            RuleFor(client => client.LastName).NotEmpty().MaximumLength(64);
+
+            RuleFor(client => client.Email).NotEmpty().MaximumLength(128);
+            
+            RuleFor(client => client.Password).NotEmpty().MaximumLength(128);
+        });
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Common/Validators/ClientValidator.cs:20]: Add `OnUpdate` rule set

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Commands/ClientUpdateCommand.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+using Feedback.Analyzer.Domain.Entities;
+
+namespace Feedback.Analyzer.Application.Clients.Commands;
+
+/// <summary>
+/// Represents a command to update a client entity.
+/// </summary>
+public class ClientUpdateCommand : ICommand<Client>
+{
+    public Client? Client { get; set; }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Commands/ClientUpdateCommand.cs:11]: Change to not nullable

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Commands/ClientGetCommand.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+﻿using Feedback.Analyzer.Application.Clients.Model;
+using Feedback.Analyzer.Domain.Common.Commands;
+using Feedback.Analyzer.Domain.Entities;
+
+namespace Feedback.Analyzer.Application.Clients.Commands;
+
+/// <summary>
+/// Represents a command to retrieve a collection of clients based on specified filtering criteria.
+/// </summary>
+public class ClientGetCommand : ICommand<IQueryable<Client>>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Commands/ClientGetCommand.cs:10]: Change to Query and it should return `ICollection`

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Commands/ClientGetCommand.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+{
+    public ClientFilter? ClientFilter { get; set; }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Commands/ClientGetCommand.cs:12]: Change to not nullable

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Commands/ClientGetByIdCommand.cs [hunk 10]
```diff
@@ -0,0 +1,12 @@
+﻿using Feedback.Analyzer.Domain.Common.Commands;
+using Feedback.Analyzer.Domain.Entities;
+
+namespace Feedback.Analyzer.Application.Clients.Commands;
+
+/// <summary>
+/// Represents a command to retrieve a client by their unique identifier.
+/// </summary>
+public class ClientGetByIdCommand : ICommand<Client?>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Commands/ClientGetByIdCommand.cs:9]: Change to query

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Models/DTOs/ClientDto.cs [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+    /// <summary>
+    /// Gets or sets the last name of the client.
+    /// </summary>
+    public string LastName { get; set; } = default!;
+
+    /// <summary>
+    /// Gets or sets the email address of the client.
+    /// </summary>
+    public string Email { get; set; } = default!;
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Models/DTOs/ClientDto.cs:23]: Change to `EmailAddress`

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+        var client2 = new Client()
+        {
+            FirstName = "Bob",
+            LastName = "Richard",
+            Email = "tastBobRichard@gmail.com",
+            Password = "asdf1234"
+        };
+
+        await dbContext.Clients.AddAsync(client1);
+        await dbContext.Clients.AddAsync(client2);
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs:48]: Add them to list and use `AddRange`

**Review by sulton-max — CHANGES_REQUESTED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Commands/ClientUpdateCommand.cs [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+
+namespace Feedback.Analyzer.Application.Clients.Commands;
+
+/// <summary>
+/// Represents a command to update a client entity.
+/// </summary>
+public class ClientUpdateCommand : ICommand<Client>
+{
+    public Client Client { get; set; } = default!;
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Commands/ClientUpdateCommand.cs:12]: Change to `ClientDto`

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            FirstName = "Bob",
+            LastName = "Richard",
+            EmailAddress = "tastBobRichard@gmail.com",
+            Password = "asdf1234"
+        };
+
+        List<Client> clients = new ();
+        clients.Add(client1);
+        clients.Add(client2);
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs:49]: Just initialize within the list

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Queries/ClientGetByIdQuery.cs [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+using Feedback.Analyzer.Domain.Entities;
+
+namespace Feedback.Analyzer.Application.Clients.Queries;
+
+/// <summary>
+/// Represents a command to retrieve a client by their unique identifier.
+/// </summary>
+public class ClientGetByIdQuery : IQuery<Client>
+{
+    public Guid ClientId { get; set; }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Queries/ClientGetByIdQuery.cs:12]: Change to `ClientDto`

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Queries/ClientGetQuery.cs [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace Feedback.Analyzer.Application.Clients.Queries;
+
+/// <summary>
+/// Represents a command to retrieve a collection of clients based on specified filtering criteria.
+/// </summary>
+public class ClientGetQuery : IQuery<ICollection<Client>>
+{    
+     public ClientFilter ClientFilter { get; set; } = default!;
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Clients/Queries/ClientGetQuery.cs:14]: Return `ClientDto`

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Extensions/LinqExtensions.cs [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+        ArgumentNullException.ThrowIfNull(predicate);
+
+        if(cancellationToken.IsCancellationRequested) return default;
+        
+        foreach (var item in source)
+            if (await predicate(item))
+                return item;
+
+        return default;
+    }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Extensions/LinqExtensions.cs:32]: Add docs

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

**Review by rashidovasadbek — DISMISSED** [review]

**Review by sulton-max — DISMISSED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

**Review by rashidovasadbek — APPROVED** [review]

**Review by botirjonabdumannonovv — DISMISSED** [review]

**Review by sulton-max — APPROVED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

**Review by rashidovasadbek — APPROVED** [review]

**Pull request merged**